src/main/java/org/opensearch/flowframework/common/FlowFrameworkSettings.java

```diff
@@ -44,4 +44,14 @@ private FlowFrameworkSettings() {}
         Setting.Property.NodeScope,
         Setting.Property.Dynamic
     );
+
+    /** This setting sets the maximum number of transport request retries */
+    public static final Setting<Integer> MAX_REQUEST_RETRY = Setting.intSetting(
+        "plugins.flow_framework.max_request_retry",
+        5,
+        0,
+        20,
+        Setting.Property.NodeScope,
+        Setting.Property.Dynamic
```

Comment:
One last comment. I have looked into couple of retry settings in OpenSearch and they are not dynamic. The reason can be to avoid updating it and letting the users affect the cluster by increasing it. So let's not make it dynamic.

Reply:
The whole point of adding the setting was to enable users to update the retry amount though. I can make it non-dynamic, but it defeats the purpose of having a setting all together

Reply:
It should be dynamic.

Reply:
Settings in OpenSearch codebase is not just for dynamic updates. We use settings to keep constants like max, min value at one place. What I meant with the above comment was to just remove `Setting.Property.Dynamic` from the settings.

Let's keep it dynamic for now but if required we can change it later. Resolving the comment.

Reply:
If it's not dynamic (or index) the settings update consumer won't work... :)